log.go

```diff
@@ -0,0 +1,212 @@
+package sentry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/getsentry/sentry-go/attribute"
+)
+
+// LogLevel marks the severity of the log event.
+type LogLevel string
+
+const (
+	LogLevelTrace   Level = "trace"
+	LogLevelDebug   Level = "debug"
+	LogLevelInfo    Level = "info"
+	LogLevelWarning Level = "warning"
+	LogLevelError   Level = "error"
+	LogLevelFatal   Level = "fatal"
+)
+
+const (
+	LogSeverityTrace   int = 1
+	LogSeverityDebug   int = 5
+	LogSeverityInfo    int = 9
+	LogSeverityWarning int = 13
+	LogSeverityError   int = 17
+	LogSeverityFatal   int = 21
+)
+
+var mapTypesToStr = map[attribute.Type]string{
+	attribute.INVALID: "",
+	attribute.BOOL:    "boolean",
+	attribute.INT64:   "integer",
+	attribute.FLOAT64: "double",
+	attribute.STRING:  "string",
+}
+
+// sentryLogger implements a custom logger that writes to Sentry.
+type sentryLogger struct {
+	client  *Client
+	options LoggerOptions
+}
+
+type LoggerOptions struct {
+	// BeforeSendLog is called before a log event is sent to Sentry.
+	// Use it to mutate the log event or return nil to discard.
+	BeforeSendLog func(log *Log) *Log
```

Comment:
This should be a client option - we can also give it an experimental prefix.

Reply:
Signature should be `BeforeSendLog func(event *Event, hint *EventHint) *Event`.